Automattic/jetpack
# Instant Search: change search request for widget filters to just before rendering the widgets

Partially fixes #22231

#### Changes proposed in this Pull Request:
A search request is sent to Instant Search index for the displaying of filters in the JP search widgets.
![image](https://user-images.githubusercontent.com/1425433/149074352-7de84b54-3314-4cc6-83fb-502a7fa398de.png)

The request is loaded by hooking `parse_query` filter:
https://github.com/Automattic/jetpack/blob/ae52b7774fbb5d361681a14091a982f4eb5cbe05/projects/plugins/jetpack/modules/search/class-jetpack-instant-search.php#L82

The PR proposes to load the search result just before rendering the JP search widgets, which reduces unnecessary loads to the backend.

#### Jetpack product discussion
p1641409510448600-slack-opers

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
* Navigate to a site with Jetpack Search subscription
* Add JP search widget to a page and add filters
* Ensure filters render correctly
* Ensure if there are no JP search widgets, then no search requests are made ( will add the method to test soon.. )
